x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -468,7 +474,8 @@ private void createApiKeyAndIndexIt(
                     request.getRoleDescriptors(),
                     request.getType(),
                     ApiKey.CURRENT_API_KEY_VERSION,
-                    request.getMetadata()
+                    request.getMetadata(),
+                    featureService.clusterHasFeature(clusterService.state(), SecuritySystemIndices.SECURITY_ROLES_METADATA_MIGRATED)
```

Comment:
I think I'd keep away from touching the metadata of API key role descriptors.
Maybe we need it (I'm not sure we do), but I would prefer we simplify this PR by not handling it in this PR.

Reply:
This is needed to prevent METADATA_FLATTENED to be written to role descriptors when the cluster doesn't yet support parsing of it.